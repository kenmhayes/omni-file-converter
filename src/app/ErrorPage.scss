.cover-container {
    max-width: 42em;
}

@import "../assets/scss/bootstrapcustom.scss";