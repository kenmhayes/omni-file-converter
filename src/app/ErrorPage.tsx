import React from 'react';
import './ErrorPage.scss';
import { useRouteError } from 'react-router-dom';

/**
 * Interface for the route error object from react-router-dom package
 */
interface RouteError {
  statusText?: string;
  message?: string;
}

/**
 * An element for displaying a message indicating a route error occurred
 * @returns
 *   a react element
 */
function ErrorPage() {
  const error: RouteError = useRouteError() as RouteError;

  return (
    <div className="cover-container d-flex w-100 h-100 p-3 mx-auto flex-column">
      <main className="px-3 mb-auto">
        <h1>Oops!</h1>
        <p>Sorry, an unexpected error has occurred.</p>
        <i>{error.statusText}</i>
      </main>
    </div>
  );
}

export default ErrorPage;
