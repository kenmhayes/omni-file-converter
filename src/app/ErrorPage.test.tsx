import React from 'react';
import { create } from 'react-test-renderer';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import ErrorPage from './ErrorPage';

it('renders an errorpage with a not found message', () => {
  //data MemoryRouter is necessary for errors to work

  const routes = [
    {
      path: "/",
      element: <div>Test</div>,
      errorElement: <ErrorPage/>
    },
  ];

  const router = createMemoryRouter(routes, {
    initialEntries: ["/", "/fake"],
    initialIndex: 1,
  });

  const component = create(
    <RouterProvider router={router} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
