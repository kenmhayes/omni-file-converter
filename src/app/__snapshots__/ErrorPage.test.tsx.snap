// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an errorpage with a not found message 1`] = `
<div
  className="cover-container d-flex w-100 h-100 p-3 mx-auto flex-column"
>
  <main
    className="px-3 mb-auto"
  >
    <h1>
      Oops!
    </h1>
    <p>
      Sorry, an unexpected error has occurred.
    </p>
    <i>
      Not Found
    </i>
  </main>
</div>
`;
